tests/integration/targets/azure_rm_loganalyticsworkspace/tasks/main.yml

```diff
@@ -160,13 +161,213 @@
+            - Processor(*)\\% Idle Time
+    destinations:
+      log_analytics:
+        - workspace_resource_id: "{{ log_analytics_workspace_output.id }}"
+          name: "{{ log_analytics_workspace_output.name }}"
+    data_flows:
+      - destinations:
+          - "{{ log_analytics_workspace_output.name }}"
+        streams:
+          - Microsoft-Perf
+    append_tags: false
+    tags:
+      TestTag1: TestValue1
+  register: output
+
+- name: Assert the data collection rule is already created
   ansible.builtin.assert:
     that:
       - not output.changed
 
+- name: Add tags to data collection rule
+  azure.azcollection.azure_rm_monitordatacollectionrules:
+    name: "{{ name_rpfx }}-DCR"
+    resource_group: "{{ resource_group }}"
+    append_tags: true
+    tags:
+      TestTag2: TestValue2
+  register: output
+
+- name: Assert the data collection rule tags are updated
+  ansible.builtin.assert:
+    that:
+      - output.changed
+      - output.datacollectionrule.tags.TestTag1 == 'TestValue1'
+      - output.datacollectionrule.tags.TestTag2 == 'TestValue2'
+
+- name: Replace tags of data collection rule
+  azure.azcollection.azure_rm_monitordatacollectionrules:
+    name: "{{ name_rpfx }}-DCR"
+    resource_group: "{{ resource_group }}"
+    append_tags: false
+    tags:
+      TestTag3: TestValue3
+  register: output
+
+- name: Assert the data collection rule tags are replaced
+  ansible.builtin.assert:
+    that:
+      - output.changed
+      - output.datacollectionrule.tags.TestTag1 is not defined
+      - output.datacollectionrule.tags.TestTag2 is not defined
+      - output.datacollectionrule.tags.TestTag3 == 'TestValue3'
+
+- name: Use info module to gather information on data collection rule (direct match)
+  azure.azcollection.azure_rm_monitordatacollectionrules_info:
+    name: "{{ name_rpfx }}-DCR"
+    resource_group: "{{ resource_group }}"
+  register: output
+
+- name: Assert that data collection rule is present (direct match)
+  ansible.builtin.assert:
+    that:
+      - output.datacollectionrules.0.name == name_rpfx + '-DCR'
+
+- name: Use info module to gather information on data collection rules in resource group (list resource group)
+  azure.azcollection.azure_rm_monitordatacollectionrules_info:
+    resource_group: "{{ resource_group }}"
+  register: output
+
+- name: Assert that data collection rule is present (list resource group)
+  ansible.builtin.assert:
+    that:
+      - name_rpfx + '-DCR' in (output.datacollectionrules | community.general.json_query('[*].name'))
+
+- name: Use info module to gather information on data collection rules in subscription (list subscription)
+  azure.azcollection.azure_rm_monitordatacollectionrules_info:
+  register: output
+
+- name: Assert that data collection rule is present (list subscription)
+  ansible.builtin.assert:
+    that:
+      - name_rpfx + '-DCR' in (output.datacollectionrules | community.general.json_query('[*].name'))
```

Comment:
Can this statement be replaced in other ways?  Because currently we have an automated detection environment that prompts that the method cannot be found when calling the methods of 'ansible.builtin. json_query', 'community.general. json_query' or' json_query'.  We are still thinking of ways to solve it.  Thank you!